### PR TITLE
qemu_guest_agent: Recovery test env after some cases

### PR DESCRIPTION
**Pull request by 6-dehan**
Backup '/etc/sysconfig/qemu-ga' ahead and recover
it after the related cases is done.

ID: 1808965
Signed-off-by: Dehan Meng <demeng@redhat.com>

### CODE REVIEW

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 1]
```diff
@@ -11,6 +11,8 @@
     gagent_pkg_check_cmd = "rpm -q qemu-guest-agent"
     black_list_change_cmd = "sed -i 's/%s//g' /etc/sysconfig/qemu-ga"
     black_list_check_cmd = "grep -nr 'BLACKLIST_RPC=.*%s' /etc/sysconfig/qemu-ga"
+    black_list_backup = "/bin/cp -rf /etc/sysconfig/qemu-ga /etc/sysconfig/qemu-ga-bk"
```
Comment (vivianQizhu) [inline qemu/tests/cfg/qemu_guest_agent.cfg:14]: Why `-r`?

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 2]
```diff
@@ -11,6 +11,8 @@
...
+￼   recovery_black_list = "mv -f /etc/sysconfig/qemu-ga-bk /etc/sysconfig/qemu-ga"
```
Comment (vivianQizhu) [inline qemu/tests/cfg/qemu_guest_agent.cfg:15]: Better make your backup path a variable.

**Review by vivianQizhu — COMMENTED** [review]
I don't really understand the positions you add `_change_bl_back`, why it does not work in pairs with `_change_bl`?

**Comment by 6-dehan** [comment]
> I don't really understand the positions you add `_change_bl_back`, why it does not work in pairs with `_change_bl`?

you mean merge them into one function,right?

**Comment by vivianQizhu** [comment]
> > I don't really understand the positions you add `_change_bl_back`, why it does not work in pairs with `_change_bl`?
> 
> you mean merge them into one function,right?

No, I mean, I suppose you should add `_change_bl_back` to the end of those functions that called `_change_bl` at beginning, but seems not.

File: qemu/tests/cfg/qemu_guest_agent.cfg (hunk 1 above)
Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg:14]: just regard any file as normal file to copy,used to use it .if there is no need to add it ,I'll delete it .

File: qemu/tests/cfg/qemu_guest_agent.cfg (hunk 2 above)
Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg:15]: okay ,editing it

**Comment by 6-dehan** [comment]
> > > I don't really understand the positions you add `_change_bl_back`, why it does not work in pairs with `_change_bl`?
> > 
> > 
> > you mean merge them into one function,right?
> 
> No, I mean, I suppose you should add `_change_bl_back` to the end of those functions that called `_change_bl` at beginning, but seems not.

yeah,there are must something misunderstand you ,actually I did that you mean the _change_bl_back should be added to the end of those functions , sorry can you mark them out?

File: qemu/tests/cfg/qemu_guest_agent.cfg (hunk 1 above)
Comment (vivianQizhu) [reply qemu/tests/cfg/qemu_guest_agent.cfg:14]: `-r` is for directory but here is a file so not necessary.

**Comment by vivianQizhu** [comment]
You have `_change_bl` in `_guest_file_prepare`, `gagent_check_with_selinux`, `gagent_check_guest_exec`, `gagent_check_log`, so I suppose you should add ``_change_bl_back` for these functions.

**Comment by vivianQizhu** [comment]
Ah sorry my bad, those functions you modified are calling `_guest_file_prepare`, ignore me. Thanks.

**Comment by 6-dehan** [comment]
> Ah sorry my bad, those functions you modified are calling `_guest_file_prepare`, ignore me. Thanks.

it's okay,

File: qemu/tests/cfg/qemu_guest_agent.cfg (hunk 2 above)
Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg:15]: following is okay? or ..
in .py:
bl_path = "/etc/sysconfig/"
cmd_blacklist_backup = self.params["black_list_backup"] % (bl_path, bl_path)

in .cfg:
black_list_backup = "/bin/cp -rf %sqemu-ga %sqemu-ga-bk"
recovery_black_list = "mv -f %sqemu-ga-bk %sqemu-ga"

**Comment by 6-dehan** [comment]
@vivianQizhu

File: qemu/tests/cfg/qemu_guest_agent.cfg (hunk 2 above)
Comment (vivianQizhu) [reply qemu/tests/cfg/qemu_guest_agent.cfg]: No, I mean, you make sure the backup file is same in `black_list_backup` and `recovery_black_list`, so better define it as a variable, something like:
```
backup_file = /etc/sysconfig/qemu-ga-bk
black_list_backup = cp -f /etc/sysconfig/qemu-ga ${backup_file}
recovery_black_list = mv -f ${backup_file} /etc/sysconfig/qemu-ga

Comment (6-dehan) [reply qemu/tests/cfg/qemu_guest_agent.cfg]: okay,thanks,editing.

**Comment by vivianQizhu** [comment]
Resolve the conflict please, otherwise LGTM. And request a test result. Thanks.

**Comment by 6-dehan** [comment]
(1/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.seek.q35: PASS (60.04 s)
 (2/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.write.q35: PASS (68.10 s)
 (3/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.with_fsfreeze.q35: PASS (68.45 s)
 (4/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.with_selinux.q35: PASS (75.73 s)
 (5/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_exec.q35: PASS (78.99 s)
 (6/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist.q35: PASS (89.64 s)

**Comment by 6-dehan** [comment]
> (1/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.seek.q35: PASS (60.04 s)
> (2/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.write.q35: PASS (68.10 s)
> (3/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.with_fsfreeze.q35: PASS (68.45 s)
> (4/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.with_selinux.q35: PASS (75.73 s)
> (5/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_exec.q35: PASS (78.99 s)
> (6/6) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist.q35: PASS (89.64 s)

@vivianQizhu

**Pull request merged**